### ISSUE TITLE
Can't unset a metadata key

### ISSUE THREAD

**Issue by VeronikaMilic** [stripe/stripe-java#788]
Java: 8 
Stripe java library: 10.0.2

If I try to unset a metadata key as described [in this issue](https://github.com/stripe/stripe-java/issues/236), setting it to null using a HashMap, it does not have any effect on the value in the stripe dashboard.

I think is a stripe-java problem, because it works with curl.

Example code:

```
Map<String, String> metadata = new HashMap<>();
metadata.put("key", null);
CustomerUpdateParams customerParams = CustomerUpdateParams.builder().putAllMetadata(metadata).build()
Customer.retrieve(customer).update(customerParams);
```

**Comment by remi-stripe** [comment]
@VeronikaMilic Thanks a lot for the report. We're investigating and will follow up once we have more details (or a fix if one is needed)

**Comment by ob-stripe** [comment]
Hi, sorry for the delay! This should be fixed in version 10.9.1. You can now set `null` values in metadata maps and the values will be sent to the API instead of being ignored.
